tests/test_date_normalization.py

```
@@ -0,0 +1,101 @@
+
+def test_is_recent_job_boundary_behavior_for_recent_window(monkeypatch):
+    monkeypatch.setattr('update_jobs.datetime', _fixed_datetime_class(FIXED_NOW_UTC))
+
+    just_inside = (FIXED_NOW_UTC - timedelta(days=7) + timedelta(minutes=1)).isoformat()
+    just_outside = (FIXED_NOW_UTC - timedelta(days=7) - timedelta(minutes=1)).isoformat()
+
+    assert is_recent_job(just_inside, 7) is True
+    assert is_recent_job(just_outside, 7) is False
+
+
+def _fixed_datetime_class(fixed_now: datetime):
```

Ruff (0.15.4)

[warning] Missing return type annotation for private function `_fixed_datetime_class`

(ANN202)